tests/test_settings.py

```diff
@@ -239,6 +239,34 @@ class SettingsTestVector:
             custom_mapping_file=Path("mapping.toml"),
         ),
     ),
+    SettingsTestVector(
```

Comment:
Is it possible to add a test where a different `custom_mapping_path` is provided by both the cli and the config?

Reply:
Done -> `mixed_deps__uses_mixture_of_user_defined_identity_and_local_env_mapping` in test_packages

Reply:
That test is great, thanks!
I was thinking more of adding a test in `test_settings.py` rather than `test_packages.py` to make sure the value in cli overrides the value in config.

Reply:
That said, if it's the default behavior, I'm not sure how useful that test is. I haven't checked if we have tested that for other settings fields before.

Reply:
The test is still useful, as there may be some other ways to implement overriding defaults. There is a test that checks that `custiom_mapping_file` ovverrides default `None`: `config_file_with_mapping_and_cli__overrides_some_defaults`.
I added another one to check if the CLI value overrides config value: `config_file_with_mapping_and_cli__cli_mapping_overrides_config`